plugin/src/main/java/io/snyk/languageserver/protocolextension/SnykExtendedLanguageClient.java

```diff
@@ -209,6 +241,98 @@ public void addTrustedPaths(SnykTrustedFoldersParams param) {
 				.map(s -> s.trim()).distinct().collect(Collectors.joining(File.pathSeparator)));
 	}
 
+	@JsonNotification(value = "$/snyk.scan")
+	public void snykScan(ScanParams param) {
+		var cacheKey = new ScanInProgressKey(param.getFolderPath(), param.getProduct());
+		var scanStateHashMap = ScanState.getInstance().getScanInProgress();
+		switch(param.getStatus()) {
+		case "inProgress":
+			scanStateHashMap.put(cacheKey, true);
+			// Show scanning view state
+			break;
+		case "success":
+			scanStateHashMap.put(cacheKey, false);
+			// Show scanning finished state
+			break;
+		case "error":
+			scanStateHashMap.put(cacheKey, false);
+			// Show error state
+			break;
+		}	
+	}
+		
+	@JsonNotification(value = "$/snyk.publishDiagnostics316")
+	public CompletableFuture<?> publishDiagnostics316(PublishDiagnosticsParams param) {
+		return CompletableFuture.runAsync(() -> {
+			var snykIssueCache = SnykIssueCache.getInstance();
+			var uri = param.getUri();
+			if(StringUtils.isEmpty(uri)) {
+				return;
+			}
```

Comment:
suggestion: log this. This is an error case where the server provides data that's incomplete.